lib/awful/util.lua

```diff
@@ -253,16 +253,7 @@ end
 -- @tparam string path
 -- @treturn bool True if path exists and is a directory.
 function util.is_dir(path)
-    local file = io.open(path)
-    if file then
-        if not file:read(0) -- Not a regular file (might include empty ones).
-            and file:seek("end") ~= 0 then  -- And not a file with size 0.
-            io.close(file)
-            return true
-        end
-        io.close(file)
-    end
-    return false
+    return Gio.File.new_for_path(path):query_file_type({}) == "DIRECTORY"
```

Comment:
Try and error. After closing (well, closing is something else than freeing/unref!) I stopped getting the "all FDs used" error.

I guess some key point is the following from https://developer.gnome.org/gio/stable/GFileEnumerator.html#g-file-enumerator-close (no such text is present elsewhere):

> This will be automatically called when the last reference is dropped, but you might want to call this function to make sure resources are released as early as possible.